src/amplitude.js

```diff
@@ -13,83 +14,53 @@ var UUID = require('./uuid');
-  LAST_EVENT_ID: 'amplitude_lastEventId',
-  LAST_EVENT_TIME: 'amplitude_lastEventTime',
-  LAST_IDENTIFY_ID: 'amplitude_lastIdentifyId',
-  LAST_SEQUENCE_NUMBER: 'amplitude_lastSequenceNumber',
-  REFERRER: 'amplitude_referrer',
-  SESSION_ID: 'amplitude_sessionId',
-
-  // Used in cookie as well
-  DEVICE_ID: 'amplitude_deviceId',
-  OPT_OUT: 'amplitude_optOut',
-  USER_ID: 'amplitude_userId'
-};
-
-/*
- * Amplitude API
+/**
+ * Amplitude SDK API - instance constructor.
+ * @constructor Amplitude
+ * @public
+ * @example var amplitude = new Amplitude();
  */
-var Amplitude = function() {
+var Amplitude = function Amplitude() {
   this._unsentEvents = [];
   this._unsentIdentifys = [];
   this._ua = new UAParser(navigator.userAgent).getResult();
   this.options = object.merge({}, DEFAULT_OPTIONS);
   this.cookieStorage = new cookieStorage().getStorage();
   this._q = []; // queue for proxied functions before script load
-};
+  this._sending = false;
+  this._updateScheduled = false;
 
-Amplitude.prototype._eventId = 0;
-Amplitude.prototype._identifyId = 0;
-Amplitude.prototype._sequenceNumber = 0;
-Amplitude.prototype._sending = false;
-Amplitude.prototype._lastEventTime = null;
-Amplitude.prototype._sessionId = null;
-Amplitude.prototype._newSession = false;
-Amplitude.prototype._updateScheduled = false;
+  // event meta data
+  this._eventId = 0;
+  this._identifyId = 0;
+  this._lastEventTime = null;
+  this._newSession = false;
+  this._sequenceNumber = 0;
+  this._sessionId = null;
+};
 
 Amplitude.prototype.Identify = Identify;
 
 /**
- * Initializes Amplitude.
- * apiKey The API Key for your app
- * opt_userId An identifier for this user
- * opt_config Configuration options
- *   - saveEvents (boolean) Whether to save events to local storage. Defaults to true.
- *   - includeUtm (boolean) Whether to send utm parameters with events. Defaults to false.
- *   - includeReferrer (boolean) Whether to send referrer info with events. Defaults to false.
+ * Initializes the Amplitude Javascript SDK with your apiKey and any optional configurations.
+ * This is required before any other methods can be called.
+ * @public
+ * @param {string} apiKey - The API key for your app.
+ * @param {string} opt_userId - (optional) An identifier for this user.
+ * @param {object} opt_config - (optional) Configuration options.
+ * See [Readme]{@link https://github.com/amplitude/Amplitude-Javascript#configuration-options} for list of options and default values.
+ * @param {function} opt_callback - (optional) Provide a callback function to run after initialization is complete.
+ * @example amplitude.init('API_KEY', 'USER_ID', {includeReferrer: true, includeUtm: true}, function() { alert('init complete'); });
  */
-Amplitude.prototype.init = function(apiKey, opt_userId, opt_config, callback) {
+Amplitude.prototype.init = function init(apiKey, opt_userId, opt_config, opt_callback) {
+  if (type(apiKey) !== 'string' || utils.isEmptyString(apiKey)) {
```

Comment:
Aggressively validate input. This includes checking the type, as well as for empty strings